### ISSUE TITLE
Throw error when adding Firewall rule without permission

### ISSUE THREAD

**Issue by ruffy91** [opnsense/core#3626]
**Important notices**
Before you add a new report, we ask you kindly to acknowledge the following:

[x] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md

[x] I have searched the existing issues and I'm convinced that mine is new.

**Is your feature request related to a problem? Please describe.**
When a group has set "System: Deny config write" and adds a Firewall rule the rule is not created but the user gets no error. The normal prompt to apply rule changes is shown.

**Describe the solution you'd like**
There should be a clear and concise error (just like for example when changing a setting in the Bind Plugin) that the user has set the "System: Deny config write" permission and cannot add Firewall Rules as a consequence.

**Additional context**
When creating a new Group and adding all permissions the "System: Deny config write" permission is applied. This is unintuitive.

**Comment by AdSchellevis** [comment]
I expect this isn't an isolated event, it's just a pity that we didn't drop the feature some years ago. Preventing config writes is quite hackery and doesn't really fit normal acl behaviour. 

We added a message in the privilege section some time ago, to warn about things like this (and possible future removal).
![image](https://user-images.githubusercontent.com/9783985/62463800-3c35b280-b78b-11e9-927a-be2a2fb4716d.png)

**Comment by ruffy91** [comment]
Hello Ad,

I do not see that warning in 19.7.2, was it removed?
![image](https://user-images.githubusercontent.com/344634/62486198-e418a380-b7be-11e9-9c92-d82422e24d9c.png)

Also for example if i change a setting in the net-bind plugin I get the following warning:
![image](https://user-images.githubusercontent.com/344634/62486316-38238800-b7bf-11e9-989e-5f5c2e0e2c64.png)

**Comment by AdSchellevis** [comment]
![image](https://user-images.githubusercontent.com/9783985/62486415-85075e80-b7bf-11e9-9b4e-7877be884be3.png)
It's shown after selection of the privilege (also in 19.7.2)

**Comment by AdSchellevis** [comment]
for reference, mvc based code can throw exceptions, which isn't possible in legacy based parts of the system.

https://github.com/opnsense/core/blob/5c04e65dcf8451aa719afbec272d0e80c62a88ca/src/opnsense/mvc/app/controllers/OPNsense/Base/ApiMutableModelControllerBase.php#L188-L191

**Comment by ruffy91** [comment]
I literally just modified these settings and the popup does not appear.
I use a chromium based Browser (Edge Dev) and did disable my adblocker as an additional test.
Also in IE no popup is shown.

To reproduce: Edit Privileges of a group, deselect all, save, select all, save.
No message is shown.

Also if this feature will be removed, will there be a new possibility to restrict user to specific settings or will administration of a OPNSense Firewall be an all or nothing decision?

**Comment by AdSchellevis** [comment]
This specific permission might be removed in the future, although it might be a long term goal, new style software is easier to restrict (endpoints are usually separated). 

Fixing all legacy components to handle "read-only" correctly would probably be quite time consuming. (we inherited the feature, unfortunately it has always been as flawed as you have experienced)

If we could easily add a generic exception handler, we would probably do so, for our new mvc code this was easier todo (most components will likely serve the same error message there).

**Comment by ruffy91** [comment]
Thanks for the clarification.

As I understood the underlying mechanism that allows the "Permission denied" message is only implemented for new mvc views, so as more pages are implemented with the new backend the smaller the problem with the missing message will be.

TLDR for others having similar problems:
If you want a Admin Group which is allowed to do everything DO NOT use "Select all" but instead just give the "GUI All Pages" permission.
Also do not use "Deny config write" if possible.

**Comment by AdSchellevis** [comment]
@ruffy91 the select all issue will likely be fixed in 20.1 (https://github.com/opnsense/core/issues/3569), it's quite tricky now indeed.
